bndtools.core/src/bndtools/launch/BndDependencySourceContainer.java

```diff
@@ -98,11 +100,22 @@ public ISourceContainerType getType() {
                         }
                     }
                 }
+
+                lastRun = run;
```

Comment:
You have this lastRun which is ended in dispose. Are there cases where lastRun could be non-null here before it is set to run? In that case, should the non-null last run be ended?